### PR TITLE
Refactoring and other code improvements

### PR DESCRIPTION
**Pull request by lmcmicu**

### CODE REVIEW

**Comment by jamesaoverton** [comment]
For the OBO Technical Services grant we said that we would review and refactor the code for our shared infrastructure to facilitate long-term maintenance. I asked @lmcmicu to do this refactoring a couple of months ago, and then I dropped the ball. @rctauber and I will review this now. If @cmungall or anyone else has time to review in the next little while, please let us know.

File: util/extract-metadata.py [hunk 1]
```diff
@@ -1,242 +1,227 @@
...
-    args = parser.parse_args()
+  args = parser.parse_args()
 
-    func = args.function
-    func(args)
+  func = args.function
+  func(args)
 
 
 def validate_markdown(args):
```
Comment (jamesaoverton) [inline util/extract-metadata.py]: This job is now done better by the `validate-metadata.py` script and schemas. It isn't used in the `Makefile`. I think we should remove this function and subcommand parser.

Comment (lmcmicu) [reply util/extract-metadata.py]: Hi James (and all), thanks for having a look. FYI you probably already know this, but you can hide whitespace changes temporarily while reviewing the commit by clicking on the little settings (gear) icon on the Files Changed tab.

Comment (jamesaoverton) [reply util/extract-metadata.py]: Thanks @lmcmicu. Actually I didn't know about that option. You learn something new every day.

**Comment by lmcmicu** [comment]
Thanks, everyone, for taking a look.

**Pull request merged**